src/validation.cpp

```diff
@@ -3435,6 +3435,12 @@ bool ProcessNewBlockHeaders(const std::vector<CBlockHeader>& headers, CValidatio
         }
     }
     NotifyHeaderTip();
+    {
+        LOCK(cs_main);
+        if (IsInitialBlockDownload() && ppindex && *ppindex && pindexBestHeader) {
```

Comment:
Could add a comment why you are checking for `pindexBestHeader`, which is unused below.

Reply:
It's a relict and not necessary. Removed it just now.